### ISSUE TITLE
Is the best model determined test accuracy?

### ISSUE THREAD

**Issue by Li-Muyang** [microsoft/Semi-supervised-learning#98]
Hi, I just want to make sure whether you used validation set or test set to decide which iteration gives the best model.

**Comment by Hhhhhhao** [comment]
If dataset has validation split, it is decided on validation split. If not, test split is directly used.

**Comment by Li-Muyang** [comment]
In that case, it probably makes more sense to just use the model from latest iteration?

**Comment by Re-dot-art** [comment]
I also have doubts about this issue. As far as I know, the original paper did not have a validation set and was based on the highest accuracy obtained during training on the test set.
